src/serde/reader.rs

```diff
@@ -90,19 +90,15 @@ impl fmt::Display for DeserializeError {
                 write!(fmt, "InvalidEncoding: {}", ib),
             DeserializeError::SizeLimit =>
                 write!(fmt, "SizeLimit"),
-            DeserializeError::Serde(ref s) =>
-                s.fmt(fmt),
+            DeserializeError::Custom(ref s) =>
+                write!(fmt, "Custom Error {}", s),
```

Comment:
A major focus of serde 0.9 is better error message so as the person doing this upgrade, part of your job is improving existing bad error messages!